src/ocl/batchnormocl.cpp

```diff
@@ -160,15 +160,26 @@ void BatchNormForwardTraining(Handle& handle,
         unsigned int ldsgcn   = xlocalsize / 64;
         unsigned int ldsnogcn = xlocalsize;
         std::string algo_name = "miopenBatchNormForwardTrainingSpatial";
-        if((in_nhw < 33554432 && in_cstride > 1024) || ((in_cstride > 60) && bfpmixparm))
+
+        if(n < 3)
+        {
+            variant    = 4;
+            xlocalsize = 512;
+            xgridsize  = c * xlocalsize;
+            ylocalsize = 1;
+            ygridsize  = 1;
+            ldsgcn     = 8;
+            ldsnogcn   = 512;
+        }
+        else if((in_nhw < 33554432 && in_cstride > 1024) || ((in_cstride > 60) && bfpmixparm))
         {
             //
         }
         else if(in_cstride <= 512)
         {
             variant = 0;
         }
-        else
+        else if(((n > 768) && (in_cstride > 150) && bfp32parm) || in_nhw >= 33554432)
```

Comment:
Do we need in_nhw >= 33554432 here?  I don't see that in original code..

Reply:
Compressed the logic. Please check if it amounts to the same as before. I’ll double check.

Reply:
It seems fine to me.  If your performance testing is good,  Can you please merge it?